components/nexus-httpclient/src/main/java/org/sonatype/nexus/httpclient/config/ConfigurationCustomizer.java

```diff
@@ -88,7 +89,12 @@ private void apply(final ConnectionConfiguration connection, final HttpClientPla
       plan.getClient().setRetryHandler(new StandardHttpRequestRetryHandler(connection.getMaximumRetries(), false));
     }
 
-    // TODO: user-agent (and/or custom headers?)
+    if (connection.getUserAgentCustomisation() != null) {
+      // append UA customisation to one stored as attribute by DefaultsCustomizer
+      plan.getHeaders().put(HttpHeaders.USER_AGENT,
+          plan.getAttributes().get(HttpHeaders.USER_AGENT) + " " + connection.getUserAgentCustomisation()
+      );
+    }
```

Comment:
USER_AGENT header is already, set, any reason why we don't plan.getHeaders().get(HttpHeaders.USER_AGENT) vs. attributes and the above change which stuffs it into attributes?